common/static/common/js/discussion/views/discussion_topic_menu_view.js

```diff
@@ -149,44 +146,27 @@
             },
 
             // @TODO move into utils.coffee
-            getNameWidth: function(name) {
-                var $test = $('<div>'),
-                    width;
-
-                $test.css({
-                    'font-size': this.dropdownButton.css('font-size'),
-                    'opacity': 0,
-                    'position': 'absolute',
-                    'left': -1000,
-                    'top': -1000
-                }).html(name).appendTo(document.body);
-                width = $test.width();
-                $test.remove();
-                return width;
-            },
-
-            // @TODO move into utils.coffee
             fitName: function(name) {
                 var ellipsisText = gettext('…'),
                     partialName, path, rawName;
 
-                if (this.getNameWidth(name) < this.maxNameWidth) {
+                if (name.length < this.maxNameWidth) {
                     return name;
                 } else {
                     path = _.map(name.split('/'), function(item){
                         return item.replace(/^\s+|\s+$/g, '');
                     });
                     while (path.length > 1) {
                         path.shift();
-                        partialName = ellipsisText + ' / ' + path.join(' / ');
-                        if (this.getNameWidth(partialName) < this.maxNameWidth) {
-                          return partialName;
+                        partialName = ellipsisText + '/' + path.join('/');
+                        if (partialName.length < this.maxNameWidth) {
+                            return partialName;
                         }
                     }
                     rawName = path[0];
                     name = ellipsisText + ' / ' + rawName;
-                    while (this.getNameWidth(name) > this.maxNameWidth) {
-                      rawName = rawName.slice(0, -1);
+                    if (name.length + 4 > this.maxNameWidth) {
```

Comment:
Is the ``+ 4`` to handle the ellipsis text plus `` / ``? This is a little brittle since ``ellipsisText`` is determined with ``gettext`` and so a translation might make it longer. Can you change the 4 to be computed from the length of the various strings that will be concatenated.